cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -67,9 +68,10 @@
    * @param systemArgs the system arguments for this program run
    * @param sourceId id of the source of program run status, which is proportional to the timestamp of
    *                 when the current program run status is reached
+   * @param artifactId artifact id used to create the application the program belongs to
    */
   void setProvisioning(ProgramRunId id, long startTime, Map<String, String> runtimeArgs,
-                       Map<String, String> systemArgs, byte[] sourceId);
+                       Map<String, String> systemArgs, byte[] sourceId, ArtifactId artifactId);
```

Comment:
Not sure if it is better to pass `artifactId` as parameter or add them to `ProgramRunId` perhaps. open to suggestions.

Reply:
Separate